gen/schema-header.act

```diff
@@ -2465,9 +2480,67 @@ def get_path_name(dnode: DNode) -> str:
 
     # This never happens, we always get at least our name!
     if len(path) == 0:
-        raise ValueError(f"No path for DNode{dnode}")
+        raise ValueError(f"No path for DNode {dnode}")
 
     return "__".join(reversed(path))
 
+def get_path(dnode: DNode) -> str:
+    """Get the path of a data node
+
+    The path is the absolute path in the YANG model, e.g. for the following
+    YANG model:
+
+    module foo {
+      prefix foo;
+      container bar {
+        leaf baz {
+          type string;
+        }
+      }
+    }
+
+    the path for the baz leaf is "/foo:bar/baz".
+
+    This function detects conflicting names across sibling nodes. If we add
+    another module that augments a conflicting node like this:
+
+    module m2 {
+      import foo { prefix foo; }
+      augment /foo:bar {
+        leaf baz {
+          type string;
+        }
+      }
+    }
+
+    Then the path for the baz leaves in the two modules is:
+    - /foo:bar/foo:baz
+    - /foo:bar/m2:baz
+    """
+    path = []
+    parent = dnode.parent
+    if parent is None and isinstance(dnode, DNodeInner):
+        top_solo = True if not dnode.children else False
+    else:
+        top_solo = False
+    while True:
+        self_name = dnode.name
+        parent = dnode.parent
+        # if parent is not None then of course it is DNodeInner (= has children
+        # attribute), but let's make that explicit for the compiler ...
+        if parent is not None:
+            if isinstance(parent, DNodeInner):
```

Comment:
This is redundant. `if isinstance...` implies that it is not None. I fixed this earlier this year, before `isinstance` would crash if the value was None, so you indeed had to write what you had here, not to make the compiler happy but to avoid segfault at runtime.